### ISSUE TITLE
`custom-error-definition` false positive on constant (no-substitution) template literal for name

### ISSUE THREAD

**Issue by TomerAberbach** [sindresorhus/eslint-plugin-unicorn#3378]
`repro.js`:

```js
// Case A: constant template literal assigned in constructor (false positive)
class FooError extends Error {
	constructor(message) {
		super(message);
		this.name = `FooError`;
	}
}

// Case B: constant template literal as a class field (false positive)
class BarError extends Error {
	name = `BarError`;
	constructor(message) {
		super(message);
	}
}
```

`eslint.config.js`:

```js
import unicorn from 'eslint-plugin-unicorn';

export default [
	{
		plugins: {unicorn},
		rules: {'unicorn/custom-error-definition': 'error'},
	},
];
```

Actual output:
```
  5:15  error  The `name` property should be set to `FooError`  unicorn/custom-error-definition
  11:9  error  The `name` property should be set to `BarError`  unicorn/custom-error-definition
```

Expected output: No errors. `` `FooError` `` and `'FooError'` are equivalent constant strings, so a no-substitution template literal should be accepted just like a string literal.
